src/sklearn_evaluation/plot/precision_recall.py

```diff
@@ -4,148 +4,332 @@
+
+
+def _precision_recall_metrics_multiclass(y_true, y_score):
+    precision, recall, _ = precision_recall_curve(y_true.ravel(), y_score.ravel())
+    average_precision = average_precision_score(y_true, y_score, average="micro")
+    return precision, recall, average_precision
+
+
+def _metric_list(y_true, y_score, n_classes):
+    y_true_bin = label_binarize(y_true, classes=np.unique(y_true))
+
+    plot_metrics_recall, plot_metrics_precision, plot_metrics_avg_precision = (
+        [],
+        [],
+        [],
+    )
+
+    (
+        precision,
+        recall,
+        micro_avg_precision,
+    ) = _precision_recall_metrics_multiclass(y_true_bin, y_score)
+
+    plot_metrics_recall.append(recall)
+    plot_metrics_precision.append(precision)
+    plot_metrics_avg_precision.append(micro_avg_precision)
+
+    for i in range(n_classes):
+        precision, recall, avg_precision = _precision_recall_metrics(
+            y_true_bin[:, i], y_score[:, i]
+        )
+        plot_metrics_recall.append(recall.tolist())
+        plot_metrics_precision.append(precision.tolist())
+        plot_metrics_avg_precision.append(avg_precision)
+
+    return plot_metrics_recall, plot_metrics_precision, plot_metrics_avg_precision
+
+
+def _plot_metrics_multiclass(recall, precision, labels, ax):
+    for r, p, l in zip(recall, precision, labels):
+        ax.plot(r, p, label=l)
+    return ax
+
+
+def _plot_metrics_binary(recall, precision, label, ax):
+    ax.plot(recall, precision, label=label)
+    return ax
+
+
+class PrecisionRecall(AbstractPlot):
+    """
+    Plot precision recall curve.
 
     Examples
     --------
-    .. plot:: ../examples/precision_recall.py
+    .. plot:: ../examples/precision_recall_oop.py
 
+    Notes
+    -----
+    .. versionchanged:: 0.9
     """
-    if any((val is None for val in (y_true, y_score))):
-        raise ValueError("y_true and y_score are needed to plot " "Precision-Recall")
-
-    if ax is None:
-        ax = plt.gca()
-
-    # get the number of classes from y_score
-    y_score_is_vector = is_column_vector(y_score) or is_row_vector(y_score)
-    if y_score_is_vector:
-        n_classes = 2
-    else:
-        _, n_classes = y_score.shape
-
-    # check data shape?
-
-    if n_classes > 2:
-        # convert y_true to binary format
-        y_true_bin = label_binarize(y_true, classes=np.unique(y_true))
-        _precision_recall_multi(y_true_bin, y_score, ax=ax)
-        for i in range(n_classes):
-            _precision_recall(y_true_bin[:, i], y_score[:, i], ax=ax)
-    else:
+
+    @SKLearnEvaluationLogger.log(feature="plot", action="precision-recall-init")
+    def __init__(self, recall, precision, average_precision, label=None):
+        self.recall = recall
+        self.precision = precision
+        self.average_precision = average_precision
+        self.label = label
+
+        _validate_metrics_input(self.recall, self.precision, self.average_precision)
+
+    def plot(self, ax=None):
```

Comment:
I think we're missing telemetry here?

Reply:
I think we're only adding telemetry to the constructor, the plot method is called internally, so no need to add it here.